http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -74,31 +69,30 @@ void addListener(final WebSocketConnectionListener listener) {
 
   /** Does an async close of the current websocket connection. */
   void close() {
-    if (client.isOpen()) {
-      new Thread(
-              () -> {
-                if (client.isOpen()) {
-                  client.getConnection().close();
-                }
-              })
-          .start();
-    }
     closed = true;
+    client.close();
   }
 
+  /**
+   * Initiates a websocket connection. Must be called before {@link #sendMessage(String)}. This
+   * method blocks until the connection has either successfully been established, or the connection
+   * failed.
+   */
   void connect() {
     Preconditions.checkState(!client.isOpen());
     Preconditions.checkState(!closed);
-    webSocketConnector.initiateConnection();
+    Interruptibles.await(
```

Comment:
The return value of the interruptible is not used. If we go on to call `send`, we may get an odd exception. Do you have any thoughts about this error-case scenario and how to handle it? Perhap same considerations for when we fail to establish a connection.